pkg/geoipupdate/config.go

```diff
@@ -98,20 +102,14 @@ func WithOutput(val bool) Option {
 	}
 }
 
-// NewConfig parses the configuration file.
-// flagOptions is provided to provide optional flag overrides to the config
-// file.
-func NewConfig( //nolint: gocyclo // long but breaking it up may be worse
+// NewConfig create a new configuration and populates it based on a config
+// file point to by 'path', then by various environment variables, and then
+// finally by flag overrides provided by flagOptions. Values from the later
+// override the former.
+func NewConfig(
 	path string,
```

Comment:
I think it would be nice if we could make `path` not required since users may provide their entire configuration without using a config file

Reply:
If we want to go ahead with a major version release I could add this change to this PR

Reply:
Yeah, I'm not sure. Perhaps we should go ahead and do that. It is kind of odd how things are right now.